plugins/web-terminal/src/components/NamespaceModalComponent/NamespaceModalComponent.tsx

```diff
@@ -0,0 +1,108 @@
+import {
+  Dialog,
+  DialogTitle,
+  DialogContent,
+  DialogContentText,
+  FormControl,
+  Select,
+  MenuItem,
+  makeStyles,
+  Button,
+} from '@material-ui/core';
+import React, { useEffect } from 'react';
+import { getNamespaces } from '../TerminalComponent/utils';
+
+const useStyles = makeStyles({
+  container: {
+    display: 'flex',
+    flexWrap: 'wrap',
+    justifyContent: 'center',
+    marginBottom: '10px',
+  },
+  buttonmargin: {
+    marginTop: '10px',
+    marginBottom: '10px',
+  },
+});
+
+type props = {
+  token: string;
+  link: string;
+  open: boolean;
+  previousNamespace: string;
+  handleClose: () => void;
+  onSubmit: (selectedNamesapce: string) => void;
+};
+
+export const NamespaceModalComponent = ({
+  token,
+  link,
```

Comment:
Instead of passing token and link to the component, you can pass a callback 

aka:
```
<NamespaceModalComponent onInit={() => getNamespaces(link, token)} />
```